getLocationLambda.js

```diff
@@ -24,179 +16,183 @@ const dynamo = new AWS.DynamoDB.DocumentClient();
+        const query = event.queryStringParameters
         
-            body = {RecreationArea: body[0].RecreationArea,
-                    About: body[0].About,
-                    Images: body[0].Images,
-                    List : body
-                    }
-                    
-
-        } 
-        else if (recreationAreaFilter !== null && recreationAreaFilter !== undefined){
-            body = body.filter(location => location.RecreationArea == recreationAreaFilter)
-            body = body.map(item => {
-                item = {...item, ParkingData: 
+        try {
+            result = await dynamo.scan({ 
+                TableName: "Locations"
+            }).promise();
+            
+            result = result.Items;
+            
+            let parkingData = await getParkingData()
+            
+            console.log(parkingData)
+            
+            if (query.parkingLotNameFilter !== null && query.parkingLotNameFilter !== undefined){
+                result = result.filter(location => location.ParkingLotName == query.parkingLotNameFilter)
+                result = result.map(item => {
+                    item = {...item, ParkingData:
+                                            parkingData
+                                            .sort((a, b) => b.LastUpdate - a.LastUpdate)
+                                            .filter(parkingLot => parkingLot.ParkingLotId == item.Id)}
+                    return item
+                })
+            } 
+            else if (query.parkUrlFilter !== null && query.parkUrlFilter !== undefined){
+                result = result.filter(location => location.ParkURL == query.parkUrlFilter);
+                result = result.map(item => {
+                    item = {...item, ParkingData: 
                                         parkingData
                                         .sort((a, b) => b.LastUpdate - a.LastUpdate)
                                         .filter(parkingLot => parkingLot.ParkingLotId == item.Id)}
-                return item
+                    return item
                 })
-            body = {RecreationArea: body[0].RecreationArea,
-                    About: body[0].About,
-                    Images: body[0].Images,
-                    List : body
-                    }
-        }
-        else if  (countyFilter !== null && countyFilter !== undefined && countyFilter === "all"){
-            var counties = [];
-            for(var i = 0; i<body.length;i++){
-                counties.push(body[i].County)
+            
+                result = {RecreationArea: result[0].RecreationArea,
+                        About: result[0].About,
+                        Images: result[0].Images,
+                        List : result
+                        }
+                        
+    
+            } 
+            else if (query.recreationAreaFilter !== null && query.recreationAreaFilter !== undefined){
```

Comment:
TODO:
You can just do if(countyFilter) instead of checking for null, undefined, etc

https://stackoverflow.com/questions/5515310/is-there-a-standard-function-to-check-for-null-undefined-or-blank-variables-in